keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ResourceStatusService.kt

```diff
@@ -103,15 +103,46 @@ class ResourceStatusService(
   }
 
   /**
-   * Checks last 10 events for flapping between only ResourceActuationLaunched and ResourceDeltaDetected
+   * Returns true if a resource has been vetoed by the unhappy veto,
+   * or if the last 10 events are only ResourceActuationLaunched or ResourceDeltaDetected events,
+   * or if the resource has been vetoed by an unspecified veto that we don't have an explicit status mapping for.
    */
   private fun List<ResourceHistoryEvent>.isUnhappy(): Boolean {
+    if (first() is ResourceActuationVetoed && (first() as ResourceActuationVetoed).isUnhappy()) {
+      return true
+    }
+
     val recentSliceOfHistory = this.subList(0, Math.min(10, this.size))
     val filteredHistory = recentSliceOfHistory.filter { it is ResourceDeltaDetected || it is ResourceActuationLaunched }
-    if (filteredHistory.size != recentSliceOfHistory.size) {
-      // there are other events, we're not thrashing.
-      return false
+    if (filteredHistory.size == recentSliceOfHistory.size) {
+      return true
     }
-    return true
+
+    // a catch all for vetoes that don't have an explicit status so they don't get unknown status.
+    return first() is ResourceActuationVetoed
+  }
+
+  /**
+   * Looks at the veto event and determines if it was vetoed by the UnhappyVeto.
+   * Looks at both the [vetoer] and the [reason] for backwards compatibility.
+   */
+  private fun ResourceActuationVetoed.isUnhappy(): Boolean =
+    vetoer?.contains("unhappy", true) ?: false || reason?.contains("unhappy", true) ?: false
```

Comment:
`vetoer` is a new field, which defaults to `null`. In order to be backwards compatible we look at both the `vetoer` or we parse the `reason`.